client/rpc-api/src/policy.rs

```diff
@@ -56,7 +56,7 @@ impl std::fmt::Display for UnsafeRpcError {
 impl std::error::Error for UnsafeRpcError {}
 
 impl From<UnsafeRpcError> for rpc::Error {
-	fn from(_: UnsafeRpcError) -> rpc::Error {
-		rpc::Error::method_not_found()
+	fn from(error: UnsafeRpcError) -> rpc::Error {
+		rpc::Error { code: rpc::ErrorCode::MethodNotFound, message: error.to_string(), data: None }
```

Comment:
We should really emit another error code for this.

The method is still found but denied because it's unsafe. 
Perhaps out of scope for this PR...

Reply:
Personally I agree, although [the specs](https://www.jsonrpc.org/specification#error_object) says this:

```
-32601 	Method not found 	The method does not exist / is not available.
```

So technically this *does* fall under the "is not available". Unfortunately the specs don't define a "forbidden" error code, so we'd have to use one of the server error codes for that, which... is probably fine? I don't know how those error codes are used by the consumers of our API though; will changing this error code break anything for anyone? (Which I imagine it could, if someone's already using the current error code to detect this situation.)

But yeah, that's out-of-scope for this PR; I just wanted to make the situation at least *a little* better while I was at it.